python/ray/serve/_private/autoscaling_policy.py

```diff
@@ -46,6 +46,17 @@ def calculate_desired_num_replicas(
     smoothed_error_ratio = 1 + ((error_ratio - 1) * autoscaling_config.smoothing_factor)
     desired_num_replicas = math.ceil(current_num_replicas * smoothed_error_ratio)
 
+    # If error_ratio = 0 meaning there is no more traffic, and desired
+    # num replicas is stuck at a positive number of the math.ceil above,
+    # decrease desired_num_replicas by one so that the deployment can
+    # eventually scale to 0.
+    if (
+        error_ratio == 0
+        and desired_num_replicas == current_num_replicas
+        and desired_num_replicas >= 1
```

Comment:
Why do we check `>= 1` as opposed to `== 1`?

As written, this would continuously reduce `desired_num_replicas` by `1` whenever there's a period with no traffic, even if `num_replicas` is greater than `1`. That would be a behavior change for autoscaling when a deployment has `1` to `n` replicas. Is this intentional?

Reply:
Hmm, I don't think this is a behavior change. For instance if `smoothing_factor = 1`, our autoscaling would have downscaled directly to 0, regardless of whether `num_replicas` is currently `1` or `n`. The issue is if `smoothing_factor < 1`, our autoscaling will sometimes get stuck so we need to "help" it downscale to zero.
Checking for only `1` doesn't generalize either, for instance if `smoothing_factor = 0.2`, the number of replicas will get stuck at 4 and never go lower.

Reply:
Oh, that makes sense. Thanks for walking me through it!